packages/textlint-tester/src/textlint-tester.ts

```diff
@@ -64,54 +112,116 @@ export type TesterInvalid = {
+    });
+    return testRuleSet;
+}
+
+function createTestPluginSet(testConfigPlugins: TestConfigPlugin[]): TestPluginSet {
+    const testPluginSet: TestPluginSet = {
+        plugins: {},
+        pluginOptions: {}
+    };
+    testConfigPlugins.forEach(plugin => {
+        const pluginName = plugin.pluginId;
+        const pluginOptions = plugin.options || {};
+        testPluginSet.plugins[pluginName] = plugin.plugin;
+        testPluginSet.pluginOptions[pluginName] = pluginOptions;
+    });
+    return testPluginSet;
+}
+
 export class TextLintTester {
     constructor() {
         if (typeof coreFlags === "object") {
             coreFlags.runningTester = true;
         }
     }
 
-    testValidPattern(ruleName: string, rule: TextlintRuleCreator, valid: TesterValid) {
+    testValidPattern(name: string, param: TextlintRuleCreator | TestConfig, valid: TesterValid) {
         const text = typeof valid === "object" ? valid.text : valid;
         const inputPath = typeof valid === "object" ? valid.inputPath : undefined;
-        const options = (typeof valid === "object" && valid.options) || {};
         const ext = typeof valid === "object" && valid.ext !== undefined ? valid.ext : ".md";
         const textlint = new TextLintCore();
-        textlint.setupRules(
-            {
-                [ruleName]: rule
-            },
-            {
-                [ruleName]: options
+        if (isTestConfig(param)) {
+            const testRuleSet = createTestRuleSet(param.rules);
+            textlint.setupRules(testRuleSet.rules, testRuleSet.rulesOptions);
+            if (param.plugins !== undefined) {
+                const testPluginSet = createTestPluginSet(param.plugins);
+                textlint.setupPlugins(testPluginSet.plugins, testPluginSet.pluginOptions);
             }
-        );
+        } else {
+            const options = (typeof valid === "object" && valid.options) || {};
+            textlint.setupRules(
+                {
+                    [name]: param
+                },
+                {
+                    [name]: options
+                }
+            );
+        }
         it(inputPath || text, () => {
             return testValid({ textlint, inputPath, text, ext });
         });
     }
 
-    testInvalidPattern(ruleName: string, rule: TextlintRuleCreator, invalid: TesterInvalid) {
+    testInvalidPattern(name: string, param: TextlintRuleCreator | TestConfig, invalid: TesterInvalid) {
         const errors = invalid.errors;
         const inputPath = invalid.inputPath;
         const text = invalid.text;
-        const options = invalid.options || {};
         const ext = invalid.ext !== undefined ? invalid.ext : ".md";
         const textlint = new TextLintCore();
-        textlint.setupRules(
-            {
-                [ruleName]: rule
-            },
-            {
-                [ruleName]: options
+        if (isTestConfig(param)) {
+            const testRuleSet = createTestRuleSet(param.rules);
```

Comment:
Assert: `options` property is invalid, or ignored.

If `options` is passed, to throw an error is helpful.